csrc/scheduler/transpose.cpp

```diff
@@ -87,6 +81,12 @@ bool TransposeScheduler::canScheduleCompileTime(Fusion* fusion) {
     return false;
   }
 
+  if (!hasAtLeastTwoValidGroups(fusion)) {
```

Comment:
I moved this here because because it uses `getInputsOutputsWithInnerDim`, which uses `FindAllMappedDims`, which asserts the existence of mappings between sibling tensors.

https://github.com/NVIDIA/Fuser/blob/main/csrc/scheduler/utils.cpp#L1476

I'm not entirely sure why we should have an assertion there. If the `from` tensor has a mapping to `nullptr`, which is not asserted, but propagating that to a sibling is asserted. I think we can just remove the assertion, but not 100% sure.

Instead, we shouldn't do this level of analysis until the fusion is confirmed to consist of only pointwise ops with no reduction. This should avoid the issue and also would help reduce the segmentation cost.